.pre-commit-config.yaml

```
@@ -0,0 +1,33 @@
+# Configuration file for pre-commit (https://pre-commit.com/).
+# Please run `pre-commit run --all-files` when adding or changing entries.
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        args: [--ignore-words=.codespellignore]
+        types_or: [jupyter, markdown, python, shell]
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.950
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - click != 8.1.0
+          - stactools
```

I can't quite tell from the scripts and workflow files, but I would assume that the correct python env is activated with the necessary dev deps when pre-commit is executed, and if so, then to avoid worrying about the `rev` above being out of sync with whatever version of mypy gets installed, replace `additional_dependencies` with this instead:

```suggestion
        language: system
```

---

this needs to be fixed upstream https://github.com/stactools-packages/template/blob/main/.pre-commit-config.yaml
mypy is pinned, so these deps will work with the pinned version